### PR TITLE
bugs & ticket transactions

### PR DESCRIPTION
**Pull request by Platonenkov**
- [x] Decimal Point Error https://github.com/Transia-RnD/xrpl.c/issues/13
- [x] Hex Memo Decode https://github.com/Transia-RnD/xrpl.c/issues/15
- [x] Ticket Create https://github.com/Transia-RnD/xrpl.c/issues/16
- [x] Currency Code Hex vs. NonHex https://github.com/Transia-RnD/xrpl.c/issues/17
- [x]  Offer should be string not uint/double
- [x] Add Message to ripple client exception
- [ ] Unit/Integration Tests

### CODE REVIEW

File: Xrpl.C/Xrpl/Client/Models/Transaction/TicketCreateCreateTransaction.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
+﻿using Newtonsoft.Json;
```
Comment (dangell7) [inline Xrpl.C/Xrpl/Client/Models/Transaction/TicketCreateCreateTransaction.cs]: Can you update this function to match this? Those other fields are ledger object fields and should be on the ticket ledger object.

https://github.com/Transia-RnD/xrpl.c/blob/ad3c8a961f545ff12b192964d2749aabfe218935/Xrpl.C/Xrpl/Client/Models/Transaction/TicketCreateTransaction.cs

File: Xrpl.C/Xrpl/Client/Extensions/ExtensionHelpers.cs [hunk 2]
```diff
@@ -16,19 +19,33 @@ public static string ToHex(this string input)
...
         {
+            if (string.IsNullOrWhiteSpace(input))
+                return null;
             byte[] bytes = Enumerable.Range(0, input.Length)
                 .Where(x => x % 2 == 0)
                 .Select(x => Convert.ToByte(input.Substring(x, 2), 16))
                 .ToArray();
             return HttpUtility.HtmlEncode(Encoding.ASCII.GetString(bytes));
         }
+        public static string FromHexUtf8String(this string input)
```
Comment (dangell7) [inline Xrpl.C/Xrpl/Client/Extensions/ExtensionHelpers.cs]: Can you change the name to FromHexString and remove the function above? You are right, it should be UTF8.

File: Xrpl.C/Xrpl.C.csproj [hunk 3]
```diff
@@ -56,7 +56,6 @@
     <Folder Include="Ripple.Address.Codec\" />
     <Folder Include="Ripple.Binary.Codec\" />
     <Folder Include="Ripple.Keypairs\" />
-    <Folder Include="Xrpl\" />
```
Comment (dangell7) [inline Xrpl.C/Xrpl.C.csproj:59]: And I'm just curious why these are removed. C# isn't my expert language so I was curious.

**Comment by dangell7** [comment]
I'll merge with the above changes, however, in the future once I have refactored the library completely you would need to submit tests for some of these functions.

**Comment by Platonenkov** [comment]
> I'll merge with the above changes, however, in the future once I have refactored the library completely you would need to submit tests for some of these functions.

ok, just say what exactly

**Review by dangell7 — CHANGES_REQUESTED** [review]

File: Xrpl.C/Xrpl.C.csproj (hunk 3 above)
Comment (Platonenkov) [reply Xrpl.C/Xrpl.C.csproj:59]: this item group not need for new csproj file type type

https://github.com/Platonenkov/xrpl.CSharp/commit/ee6cf36c6ce5bc987fe708f0abe5de1684276ccb

File: Xrpl.C/Xrpl/Client/Extensions/ExtensionHelpers.cs (hunk 2 above)
Comment (Platonenkov) [reply Xrpl.C/Xrpl/Client/Extensions/ExtensionHelpers.cs]: done

File: Xrpl.C/Xrpl/Client/Models/Transaction/TicketCreateCreateTransaction.cs (hunk 1 above)
Comment (Platonenkov) [reply Xrpl.C/Xrpl/Client/Models/Transaction/TicketCreateCreateTransaction.cs]: done
https://github.com/Platonenkov/xrpl.CSharp/commit/ab4ea3851fceb9db9fad64c30e53a6bb07cbdb6f

File: Xrpl.C/Xrpl/Client/Extensions/ExtensionHelpers.cs (hunk 2 above)
Comment (Platonenkov) [reply Xrpl.C/Xrpl/Client/Extensions/ExtensionHelpers.cs]: Can I refactoring to improve code readability?
I'm a .Net developer and it pains me to look at some forms of recording...
https://github.com/Platonenkov/xrpl.CSharp/commit/1d7e1f4cfa740b0e53c40035a6d56080b1b8bf07

if it ok for you I will merge it to my dev and pull-request

File: Xrpl.C/Xrpl/Client/Models/Methods/AccountOffers.cs [hunk 4]
```diff
@@ -44,7 +44,7 @@ public class Offer
         public Currency TakerPays { get; set; }
 
         [JsonProperty("quality")]
-        public uint Quality { get; set; }
+        public double Quality { get; set; }
```
Comment (dangell7) [inline Xrpl.C/Xrpl/Client/Models/Methods/AccountOffers.cs:47]: string

**Review by dangell7 — CHANGES_REQUESTED** [review]
see the one change above.

File: Xrpl.C/Xrpl/Client/Extensions/ExtensionHelpers.cs (hunk 2 above)
Comment (dangell7) [reply Xrpl.C/Xrpl/Client/Extensions/ExtensionHelpers.cs]: I will merge this, then you merge those into dev and create another PR and I'll approve it. Just so it's clear and these changes dont get lost in a big PR.

**Review by dangell7 — APPROVED** [review]

File: Xrpl.C/Xrpl/Client/Models/Methods/AccountOffers.cs (hunk 4 above)
Comment (Platonenkov) [reply Xrpl.C/Xrpl/Client/Models/Methods/AccountOffers.cs:47]: https://github.com/Transia-RnD/xrpl.c/pull/24
done

**Pull request merged**